src/server/game/Achievements/CriteriaHandler.cpp

```diff
@@ -2305,6 +2320,8 @@ void CriteriaMgr::LoadCriteriaList()
             }
             else if (tree->ScenarioStep)
                 criteria->FlagsCu |= CRITERIA_FLAG_CU_SCENARIO;
+            else if (tree->QuestObjective)
+                criteria->FlagsCu |= CRITERIA_FLAG_CU_PLAYER;
```

Comment:
Why are you reusing PLAYER flag here instead of adding another flag? This forces you to make unrelated changes to `PlayerAchievementMgr`